openedx/core/djangoapps/notifications/views.py

```diff
@@ -625,3 +630,87 @@ def get(self, request):
             'message': 'Notification preferences retrieved',
             'data': notification_configs
         }, status=status.HTTP_200_OK)
+
+
+class NotificationPreferencesView(APIView):
+    """
+    API view to retrieve and structure the notification preferences for the
+    authenticated user.
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request):
+        """
+        Handles GET requests to retrieve notification preferences.
+
+        This method fetches the user's active notification preferences and
+        merges them with a default structure provided by NotificationAppManager.
+        This provides a complete view of all possible notifications and the
+        user's current settings for them.
+
+        Returns:
+            Response: A DRF Response object containing the structured
+                      notification preferences or an error message.
+        """
+        user_preferences_qs = NotificationPreference.objects.filter(user=request.user)
+        user_preferences_map = {pref.type: pref for pref in user_preferences_qs}
+
+        # Ensure all notification types are present in the user's preferences.
+        # If any are missing, create them with default values.
+        diff = set(COURSE_NOTIFICATION_TYPES.keys()) - set(user_preferences_map.keys())
+        missing_types = []
+        for missing_type in diff:
+            new_pref = create_notification_preference(
+                user_id=request.user.id,
+                notification_type=missing_type,
+
+            )
```

Comment:
This will add unnecessary entries for core notification types. For example, for `new_comment` and `new_response` notification types, we only need 1 notification type i.e. `core`. This code will create for both notification types